client/keys/show.go

```diff
@@ -165,20 +166,23 @@ func runShowCmd(cmd *cobra.Command, args []string) (err error) {
 func fetchKey(kb keyring.Keyring, keyref string) (*keyring.Record, error) {
 	// firstly check if the keyref is a key name of a key registered in a keyring.
 	k, err := kb.Key(keyref)
+	if (strings.HasPrefix(keyref, sdk.Bech32PrefixAccAddr)) && (len(keyref) > 6) {
+		accAddr, err := sdk.AccAddressFromBech32(keyref)
+		if err != nil {
+			return k, err
+		}
+
+		k, err = kb.KeyByAddress(accAddr)
+		return k, sdkerr.Wrap(err, "Invalid key")
+	}
```

Comment:
For me, the `keys show` command works, so I don't think we need to change anything here. @likhita-809 Did you encounter any issue with this command?

Small note: on the terra issue, they are using v0.44.5. `keys show` might be bugged on 0.44, and fixed on master.

Reply:
I made this change, just in case the key name is "cosmos", we might face an issue with keys show.